File: day2/2.py
with open("data.txt", 'r') as file:
    data = list(map(int, file.read().split(',')))


def help(data: list[int], noun: int, verb: int) -> int:
    data[1], data[2] = noun, verb
    for i in range(0, len(data), 4):
        if data[i] == 99: break
        elif data[i] == 1: data[data[i + 3]] = data[data[i+1]] + data[data[i + 2]]
        elif data[i] == 2: data[data[i + 3]] = data[data[i+1]] * data[data[i + 2]]
        else: continue
    return data[0] == 19690720

for noun in range(100):
    for verb in range(100):
        if help(list(data), noun, verb):
            print(100 * noun + verb)

File: day2/2.cpp
#include <fstream>
#include <sstream>
#include <iostream>
#include <string>
#include <vector>

bool test(std::vector<int> nums, int noun, int verb) {
    nums[1] = noun; nums[2] = verb;
    for (int i{}; i < nums.size(); i += 4) {
        if (nums[i] == 99) break;
        else if (nums[i] == 1) {
            nums[nums[i + 3]] = nums[nums[i + 2]] + nums[nums[i + 1]];
        }
        else if (nums[i] == 2) {
            nums[nums[i + 3]] = nums[nums[i + 2]] * nums[nums[i + 1]];
        }
        else break;
    }
    return nums[0] == 19690720;
} 


int main() {
    std::vector<int> nums;

    std::ifstream in("data.txt"); std::string line{}, num{};

    std::getline(in, line);
    in.close();

    std::stringstream ss(line);

    while (std::getline(ss, num, ',')) nums.emplace_back(std::stoi(num)); bool found{false};

    for (int noun{}; noun <= 99; ++noun) {
        for (int verb{}; verb <= 99; ++verb) {
            if (test(nums, noun, verb)) { found = true; std::cout << 100 * noun + verb; break; }
        }
        if (found) break;
    }

    return 0;
}